### PR TITLE
Refactor package to be imported as a single package

### PR DESCRIPTION
**Pull request by barretobrock**
The aim of this refactoring is to ultimately have a single package to import when pip installing the package to a new environment. Currently, pip installing the package results in multiple modules being created (see issue #36 for more details) 

Noting some of the major changes that took place during this refactoring:
- All files that were related to the package were moved into the `reolink_api` directory
- Package name in `setup.py` changed to `reolink_api` from `reolink-api` to allow for import in Python (hyphenated names result in error)
- Submodule files that were originally uppercase have now been made lower case
- Type hinting enforced throughout most methods & functions
    - Note that `float` is used quite extensively, as opposed to `int`. This is the recommended method as `int` is implied in `float` [more info here](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower)
    - Also note that `Dict` is used over `dict`. This generic type will allow for better-defined dictionary types in the future, if so desired.
- Added a `tests` directory to eventually establish a more robust testing routine for all submodules.
- Added two new submodules:
    - `download.py` - handles downloading motion video files
    - `motion.py` - queries camera for motion events (and their video files) during a specified window

### CODE REVIEW

**Comment by barretobrock** [comment]
@Benehiko I'll make another logic pass today just to make sure there's nothing else I didn't catch. I also need to verify that all the README items were properly transferred back to this fork. 

Feel free to reach out if you have any questions in the meantime.

File: reolink_api/config_handler.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 import yaml
+from typing import Optional, Dict
 
 
 class ConfigHandler:
     camera_settings = {}
 
     @staticmethod
-    def load() -> yaml or None:
+    def load() -> Optional[Dict]:
```
Comment (barretobrock) [inline reolink_api/config_handler.py:10]: Noting here that `Optional[]` essentially is the same as `or None`

File: reolink_api/api_handler.py [hunk 2]
```diff
@@ -69,7 +75,10 @@ def login(self) -> bool:
                 print(self.token)
                 return False
             else:
-                print("Failed to login\nStatus Code:", response.status_code)
+                # TODO: Verify this change w/ owner. Delete old code if acceptable.
```
Comment (barretobrock) [inline reolink_api/api_handler.py:78]: Pointing out this change here.

File: reolink_api/recording.py [hunk 3]
```diff
@@ -51,59 +52,67 @@ def set_recording_encoding(self,
...
 
     ###########
     # RTSP Stream
     ###########
-    def open_video_stream(self, callback=None, profile: str = "main", proxies=None):
+    def open_video_stream(self, callback: Any = None, proxies: Any = None) -> Any:
         """
         'https://support.reolink.com/hc/en-us/articles/360007010473-How-to-Live-View-Reolink-Cameras-via-VLC-Media-Player'
         Blocking function creates a generator and returns the frames as it is spawned
-        :param profile: profile is "main" or "sub"
```
Comment (barretobrock) [inline reolink_api/recording.py:80]: This was unused.

**Comment by Benehiko** [comment]
@barretobrock Looking good! I'll go through it later this weekend and do the merge. Maybe we should change the package name to `reolinkapi` for consistency - the golang package is importing as this name

**Comment by barretobrock** [comment]
@Benehiko even better! I'm wrapping up my second & final logic pass so I'll add the package new change with that.

**Comment by Benehiko** [comment]
@barretobrock I have added some changes as well to the project structure and file names. I went through the changes made and it looks good to me :) 

I have added @themoosman to just review it as well 

See commit 2b3e142fe500a9b71ba57639d418b109b89bf648

**Comment by Benehiko** [comment]
> Is `requirements.txt` a requirement for having a library on PyPI? I had initially removed it in favor of placing all the necessary reqs in with setup.py, though if this file is needed for any another reason, perhaps I should add something in `setup.py` to read from this file instead of having its own list in the file? Either way, it would be easier to maintain pinned requirements in just one place over two.

For PyPi not so much, but for other developers wanting to contribute it might be a good idea. I do agree with you about having one place to maintain the requirements. Not sure if `setup.py` fulfills both contributor and PyPi needs. We could maybe try this suggestion on [SO](https://stackoverflow.com/questions/14399534/reference-requirements-txt-for-the-install-requires-kwarg-in-setuptools-setup-py#16624700) and just add a `.` to the requirements.txt file which reads from the `setup.py` file.

**Comment by Benehiko** [comment]
> It might be cool to also set a prop like `is_logged_in` here so when we call things like `_execute_command()` `is_logged_in` would automatically be checked before running the command to ensure we're connected.
> 
> I'm not sure about whether Reolink expends any resources maintaining a connection once a user's logged in, but it does seem like a good idea to at least provide the option of deferring login. Good catch +1

Would be quite useful having a prop, the reason for the `defer_login` is so users' of the library can manage their object pool (say many cameras) without needing to request the camera anything over the network before object creation giving the user that freedom of choice.

**Comment by barretobrock** [comment]
> > Is `requirements.txt` a requirement for having a library on PyPI? I had initially removed it in favor of placing all the necessary reqs in with setup.py, though if this file is needed for any another reason, perhaps I should add something in `setup.py` to read from this file instead of having its own list in the file? Either way, it would be easier to maintain pinned requirements in just one place over two.
> 
> For PyPi not so much, but for other developers wanting to contribute it might be a good idea. I do agree with you about having one place to maintain the requirements. Not sure if `setup.py` fulfills both contributor and PyPi needs. We could maybe try this suggestion on [SO](https://stackoverflow.com/questions/14399534/reference-requirements-txt-for-the-install-requires-kwarg-in-setuptools-setup-py#16624700) and just add a `.` to the requirements.txt file which reads from the `setup.py` file.

Oh that's nice! I do agree that whatever approach should be one that most developers are already familiar with. Personally, I work mainly with requirements files, since it's easier to just swap the production requirements file with a QA version that's has more bleeding-edge versions on it. I'm not sure if this is the best approach, but in the past I've just read that file into `setup.py` as a list and pass that straight to `install_requires`, making sure to filter out empty strings and stripped lines beginning with a hash (#)

**Comment by themoosman** [comment]
I second keeping `requirements.txt` it makes it easy for developers.

**Comment by themoosman** [comment]
I just did some quick tests and everything looks good.  LGTM

**Review by themoosman — APPROVED** [review]
See my comments. 
LGTM

**Comment by Benehiko** [comment]
@barretobrock @themoosman updated pypi as well https://pypi.org/project/reolinkapi/

**Pull request merged**